cwms-data-api/src/main/java/cwms/cda/data/dto/CwmsDTOBase.java

```diff
@@ -1,12 +1,24 @@
 package cwms.cda.data.dto;
 
-import com.fasterxml.jackson.annotation.JsonProperty;
-
 import cwms.cda.api.errors.FieldException;
 /*
  * Suitable base for objects not owned by an office, like a list of locations.
  * Or parameters, or the list of offices themselves.
  */
-public interface CwmsDTOBase {
-    public  void validate() throws FieldException; // additional validation logic.
+public abstract class CwmsDTOBase {
```

Comment:
FYI, interfaces can have default methods now. even in Java 8.

Reply:
yeah.... the intention of that feature is to provide ways to modify interfaces without breaking backwards compatibility. Given our codebase is self-contained, the compatibility thing isn't an issue. Additionally, you can't mark default methods as final and we don't want the entire validate() method logic to be overwritten (until we come up with a good exception of course), but we can still provide an extension point through the validateInternal() method. Might be sticking too close to the open-closed principle there.

Reply:
Ah, that makes sense.